src/stan/services/variational/print_progress.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef STAN_SERVICES_VARIATIONAL_PRINT_PROGRESS_HPP
+#define STAN_SERVICES_VARIATIONAL_PRINT_PROGRESS_HPP
+
+#include <stan/services/io/do_print.hpp>
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+namespace stan {
+  namespace services {
+    namespace variational {
+
+      void print_progress(const int m,
```

Comment:
As an FYI, I wrote this function while changing as few lines as necessary from the equivalent `print_progress()` function in `src/stan/services/mcmc/print_progress.hpp`. This is because that function should really be refactored and this is a short term solution until all such things get generalized. So I recommend I/Alp make your suggested changes in both MCMC and variational's `print_progress()` functions or we don't touch it.